src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -117,7 +117,7 @@ public static void Bind(this IConfiguration configuration, object instance, Acti
             {
                 var options = new BinderOptions();
                 configureOptions?.Invoke(options);
-                BindInstance(instance.GetType(), instance, configuration, options);
+                instance = BindInstance(instance.GetType(), instance, configuration, options);
```

Comment:
Could you explain how this fix works?  It looks to me like you’re just modifying a local.  I wouldn’t expect that have any observable change.